<a href="https://colab.research.google.com/github/KoechCarl/public-health-vaccine-prediction/blob/main/h1n1_vaccine_prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# A Machine Learning Approach to Predicting H1N1 Vaccine Adoption

## Business Understanding

#### *Background:*

Vaccination plays a critical role in preventing the spread of infectious diseases and reducing public health risks. During outbreaks or pandemics, understanding the factors that influence whether individuals receive vaccines can help governments and health organizations design better vaccination campaigns.

The dataset used in this project comes from the **National 2009 H1N1 Flu Survey**, which collected information about individuals' demographics, health behaviors, opinions about vaccines, and whether they received the H1N1 vaccine.

Although the dataset originates from the United States, the insights and modeling approach can be useful for vaccination planning in African contexts where vaccine uptake can vary significantly across populations.

---

#### *Objectives:*

The main objective of this project is to **build a classification model that predicts whether an individual is likely to receive the H1N1 vaccine based on demographic characteristics, health behaviors, and attitudes toward vaccination.**

More specifically, the project aims to:

 1. Identify patterns in demographic and behavioral data that influence vaccine uptake.

 2. Develop a predictive model capable of classifying individuals as likely or unlikely to receive the H1N1 vaccine.

 3. Evaluate the performance of the model using appropriate classification metrics.

 4. Provide insights that can help public health organizations improve vaccination outreach strategies.

Understanding these patterns is important because **low vaccination uptake can lead to increased disease transmission, higher healthcare costs, and greater pressure on healthcare systems**.

#### *Why this matters:*
 - Predicting vaccination behavior can support more effective public health planning. If health organizations can identify populations that are less likely to vaccinate, they can target these groups with education campaigns, outreach programs, and improved access to vaccination services.

 - In regions across Africa where healthcare resources may be limited, predictive insights can help ensure that interventions are directed toward the populations that need them most.

 - During disease outbreaks, public health officials may need to prioritize communities with lower vaccine acceptance rates to prevent further spread of disease.

 ---

#### *Key StakeHolders:*
1. **The Kenya Ministry of Health**
    - Responsible for coordinating vaccination campaigns and public health initiatives.

2. **The Africa Centres for Disease Control and Prevention**
    - Supports African countries in responding to infectious disease outbreaks.

3. **International Health Organizations**
    - Organizations such as **World Health Organization** and Gavi,  **The Vaccine Alliance** work extensively across African countries to improve vaccine accessibility and coverage.

Insights from predictive models could help these organizations:

 - Identify populations at risk of low vaccination uptake

 - Design targeted communication campaigns

 - Allocate vaccination resources more effectively

 ---

 #### *Problem Description:*
Despite the availability of vaccines, not all individuals choose to receive them. Vaccine hesitancy, lack of awareness, misinformation, and access barriers can all influence vaccination decisions.

Public health authorities often face difficulty determining **which individuals or communities are least likely to receive vaccines before outbreaks occur.**

Without this knowledge:

 - Vaccination campaigns may not reach the most vulnerable populations

 - Resources may be distributed inefficiently

 - Disease outbreaks may spread more rapidly.

This project therefore seeks to answer the following key question:

**Can demographic characteristics, health behaviors, and public perceptions be used to predict whether an individual will receive the H1N1 vaccine?**

## Data Understanding

The dataset used in this project comes from the **National 2009 H1N1 Flu Survey**

For the purposes of this project, **the goal is to predict whether an individual received the H1N1 vaccine, making this a binary classification problem.**

Each row in the dataset represents **one survey respondent**, while each column represents **a feature describing the respondent’s behavior, demographics, or opinions about vaccination.**

In [1]:
# Importing necessary libraries for Data Understanding
import pandas as pd
import numpy as np

In [2]:
# Loading the datasets
features = pd.read_csv('training_set_features.csv')
labels = pd.read_csv('training_set_labels.csv')

# Merging them into one with 'respondent_id' being the common column
df = pd.merge(features, labels, on='respondent_id')

### Data Exploration
Before building machine learning models, it is important to explore the dataset to understand:

 - The structure of the data

 - The types of variables present

 - The distribution of the target variable.

This helps identify potential preprocessing steps required for modeling.

In [3]:
# Let's preview the data types for each feature
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26707 entries, 0 to 26706
Data columns (total 38 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   respondent_id                26707 non-null  int64  
 1   h1n1_concern                 26615 non-null  float64
 2   h1n1_knowledge               26591 non-null  float64
 3   behavioral_antiviral_meds    26636 non-null  float64
 4   behavioral_avoidance         26499 non-null  float64
 5   behavioral_face_mask         26688 non-null  float64
 6   behavioral_wash_hands        26665 non-null  float64
 7   behavioral_large_gatherings  26620 non-null  float64
 8   behavioral_outside_home      26625 non-null  float64
 9   behavioral_touch_face        26579 non-null  float64
 10  doctor_recc_h1n1             24547 non-null  float64
 11  doctor_recc_seasonal         24547 non-null  float64
 12  chronic_med_condition        25736 non-null  float64
 13  child_under_6_mo

In [4]:
# Let's see the number of rows and columns in the dataset
df.shape

(26707, 38)

In [5]:
# Preview the first 5 rows
df.head()

,respondent_id,h1n1_concern,h1n1_knowledge,behavioral_antiviral_meds,behavioral_avoidance,behavioral_face_mask,behavioral_wash_hands,behavioral_large_gatherings,behavioral_outside_home,behavioral_touch_face,...,rent_or_own,employment_status,hhs_geo_region,census_msa,household_adults,household_children,employment_industry,employment_occupation,h1n1_vaccine,seasonal_vaccine
0,0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,1.0,...,Own,Not in Labor Force,oxchjgsf,Non-MSA,0.0,0.0,NaN,NaN,0,0
1,1,3.0,2.0,0.0,1.0,0.0,1.0,0.0,1.0,1.0,...,Rent,Employed,bhuqouqj,"MSA, Not Principle City",0.0,0.0,pxcmvdjn,xgwztkwe,0,1
2,2,1.0,1.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,...,Own,Employed,qufhixun,"MSA, Not Principle City",2.0,0.0,rucpziij,xtkaffoo,0,0
3,3,1.0,1.0,0.0,1.0,0.0,1.0,1.0,0.0,0.0,...,Rent,Not in Labor Force,lrircsnp,"MSA, Principle City",0.0,0.0,NaN,NaN,0,1
4,4,2.0,1.0,0.0,1.0,0.0,1.0,1.0,0.0,1.0,...,Own,Employed,qufhixun,"MSA, Not Principle City",1.0,0.0,wxleyezf,emcorrxb,0,0


In [6]:
# Check for imbalance
df['h1n1_vaccine'].value_counts()

,count
h1n1_vaccine,
0,21033
1,5674


The distribution of the target variable h1n1_vaccine was examined to determine whether the dataset is balanced. The results show that 21,033 respondents (78.8%) did not receive the H1N1 vaccine, while 5,674 respondents (21.2%) reported receiving the vaccine.

This indicates a **moderately imbalanced dataset**. Class imbalance can affect model performance because a model may achieve high accuracy simply by predicting the majority class. To address this, additional evaluation metrics such as **recall, precision, and F1-score** will be used when assessing model performance.